jwst/ami/NRM_Model.py

```diff
@@ -160,48 +202,53 @@ def simulate(self, fov=None, bandpass=None, pixel=None, rotate=False,
         if self.over is None:
             self.over = over
 
+        if pixweight is not None:
+            self.over = self.pixweight.shape[0]
+
         self.phi = np.zeros(len(self.ctrs))
 
         if rotate: # this is a 'temporary' rotation of self.ctrs
             #  without altering self.ctrs
             self.rotate = rotate
             self.rotctrs = leastsqnrm.rotatevectors(self.ctrs, self.rotate)
         else:
-            self.rotctrs = self.ctrs # if initialization included a
-            #   rotation self.ctrs has been rotated
+            self.rotctrs = self.ctrs
 
         if pixel is None:
             self.pixel_sim = self.pixel
         else:
             self.pixel_sim = pixel
 
         # The polychromatic case:
-        log.debug("------Simulating Polychromatic------")
-
-        self.psf_over = np.zeros((self.over * self.fov_sim,
-                        self.over * self.fov_sim))
-
-        for w, l in self.bandpass: # w: weight, l: lambda (wavelength)
-            log.debug('weight: %s', w)
-            log.debug('wavelength: %s', l)
-            log.debug('fov: %s', self.fov_sim)
-            log.debug('over: %s', self.over)
-
-            self.psf_over += w * analyticnrm2.PSF(self.pixel_sim, \
-                        self.fov_sim, self.over, self.rotctrs, self.d, l, \
-                        self.phi, centering=self.centering, \
-                        shape=self.holeshape)
+        if hasattr(self.bandpass, '__iter__'):
+            log.debug("------Simulating Polychromatic------")
+            self.psf_over = np.zeros((self.over*self.fov_sim,
+                                      self.over*self.fov_sim))
+            for w,l in self.bandpass: # w: weight, l: lambda (wavelength)
+                self.psf_over += w*analyticnrm2.PSF(self.pixel_sim, \
+                            self.fov_sim, self.over, self.rotctrs, self.d, l, \
```

Comment:
Don't need the \ at the ends of these lines. The implied continuation within the parentheses already takes care of the continuation after the commas.